### PR TITLE
Add viewId to PointerEvents

### PR DESCRIPTION
**Pull request by goderbauer**
Follow-up to https://github.com/flutter/engine/pull/42493.

### CODE REVIEW

**Comment by goderbauer** [comment]
@dkwingsmt With this in place, you should be able to rebase https://github.com/flutter/flutter/pull/127060 and implement correct view handling, I believe?

File: packages/flutter/lib/src/gestures/events.dart [hunk 1]
```diff
@@ -2241,6 +2290,7 @@ mixin _CopyPointerPanZoomEndEvent on PointerEvent {
       device: device ?? this.device,
       position: position ?? this.position,
       embedderId: embedderId ?? this.embedderId,
+      viewId: viewId ?? this.viewId,
```
Comment (dkwingsmt) [inline packages/flutter/lib/src/gestures/events.dart:2293]: nit: Move this to the top of parameter list to keep order consistent.

File: packages/flutter/lib/src/gestures/hit_test.dart [hunk 2]
```diff
@@ -16,11 +16,16 @@ export 'events.dart' show PointerEvent;
 
 /// An object that can hit-test pointers.
 abstract interface class HitTestable {
-  /// Check whether the given position hits this object.
-  ///
-  /// If this given position hits this object, consider adding a [HitTestEntry]
-  /// to the given hit test result.
+  /// Deprecated. Use [hitTestInView] instead.
+  @Deprecated(
+    'Use hitTestInView and specify the view to hit test. '
```
Comment (dkwingsmt) [inline packages/flutter/lib/src/gestures/hit_test.dart:21]: Shall we add something like this: To keep the current behavior, use 0 for viewId for the first view.

Comment (dkwingsmt) [reply packages/flutter/lib/src/gestures/hit_test.dart:21]: Also to other copies of this method.

**Review by dkwingsmt — APPROVED** [review]
LGTM

File: packages/flutter/lib/src/gestures/events.dart (hunk 1 above)
Comment (goderbauer) [reply packages/flutter/lib/src/gestures/events.dart:2293]: Done, also fixed for the other two instances where this went out of order.

File: packages/flutter/lib/src/gestures/hit_test.dart (hunk 2 above)
Comment (goderbauer) [reply packages/flutter/lib/src/gestures/hit_test.dart:21]: Done.

**Pull request merged**